contributors/design-proposals/auth/external-secrets.md

```diff
@@ -0,0 +1,218 @@
+# Providing external secrets to Pods in Kubernetes
+
+Status: Pending
+
+Version: Alpha
+
+Implementation Owner: openshift/sig-security
+
+## Motivation
+
+Kubernetes support the Secrets mechanism where secrets are stored in
+etcd and made available in Pods and containers via `volumeMounts` or
+environment variables. In these cases, the secrets are either
+automatically created in etcd (credentials with access to API) or
+generated and stored by admins/users.
+
+In some cases though, secrets can primarily be created and stored
+in sources external to the Kubernetes cluster. For example, TLS
+certificates can be created, signed, and renewed by external
+certificate authority like FreeIPA/IdM's CA, Let's Encrypt, or AD CS,
+and they can be stored in external vault storage like FreeIPA/IdM Vault
+or HashiCorp Vault. To make them available via Kubernetes's Secrets
+mechanism, we'd need to have the certificates created beforehand and
+stored in etcd first. That might not be the ideal workflow because it
+does not support requirements like automatically renewing the
+certificate if the copy found in vault is getting close to expiration,
+or auditing the actual access of individual containerized applications
+to the secrets.
+
+It is possible to run per-Node daemon via DaemonSet with `hostPath`
+access to per-Node credentials (to authenticate to the external
+vault / secrets provider), exposing Unix socket on the filesystem of
+the Node. Using flexVolume mechanism, the directory with that Unix
+socket can then be made available to Pods and containers that need
+access to the secrets. The daemon in the DaemonSet can then use
+`SO_PEERCRED`/`SO_PEERSEC` to get the identity of the process accessing
+the service via the Unix socket, and authorize it based on the Pod
+identity and other attributes / annotations which can be retrieved
+via Kubernetes API.
+
+One drawback of this approach is the need for active client tool or
+library in each of the container, to invoke the daemon's API over
+the Unix socket to request the appropriate secret / certificate.
+
+For that reason we are looking for mechanism of exposing the secrets
+from external sources to Pods and their containers via filesystem
+interface. With that approach, the individual secrets would be seen as
+files in filesystem structure, while backed by active mechanisms
+of external secrets providers or vaults, and would only request the
+secret at the time it is actually accessed by the container.
+
+## Proposal
+
+We are proposing flexVolume plugin which would utilize FUSE. FUSE
```

Comment:
You need to establish why someone consuming this proposal cares about FUSE (which is an implementation detail).  Someone could just as easily create a flex volume on tmpfs and inject it into the container - you need to articulate why FUSE is special here.

Reply:
Would the motivation/proposal in https://github.com/kubernetes/community/pull/1147 be the wording that we might be looking for?